apps/wallet-dashboard/tests/connect.spec.ts

```diff
@@ -16,18 +16,25 @@ test.describe.serial('Wallet Connection', () => {
     });
 
     test('should connect to wallet extension', async ({ page, sharedState, extensionName }) => {
-        const { sharedContext } = sharedState;
+        const { sharedContext, wallet } = sharedState;
 
         if (!sharedContext) {
-            throw new Error('Context is not defined');
+            throw new Error('Shared context expected!');
+        }
+
+        if (!wallet.address) {
+            throw new Error('Wallet address was not set');
         }
 
         await page.goto('/', { waitUntil: 'networkidle' });
         await connectWallet(page, sharedContext, extensionName);
 
         // Verify connection was successful on dashboard
-        await page.waitForSelector('[data-testid="sidebar"]');
-        await expect(page.getByTestId('sidebar')).toBeVisible();
+        await expect(page.getByText('Start Staking')).toBeVisible({ timeout: 30_000 });
+        const truncatedWalletAddress = wallet.address.slice(0, 6) + '…' + wallet.address.slice(-4);
+        await expect(page.getByText(truncatedWalletAddress).first()).toBeVisible({
+            timeout: 30_000,
+        });
```

Comment:
i think we can simplify this by just adding, wdyt? :pray: 
```suggestion
       expect(page.getByText('My Coins')).toBeVisible({ timeout: 30_000 });
```